package/zshelf/package

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020 The Toltec Contributors
+# SPDX-License-Identifier: MIT
+
+pkgnames=(zshelf)
+pkgdesc="Z-Library browser and downloader"
+url=https://github.com/khanhas/zshelf
+pkgver=0.1.1
+section=utils
+timestamp=2021-01-31T05:25Z
+maintainer="khanhas <xuankhanh963@gmail.com>"
+license=GPL-3.0
+depends=(node)
+
+source=(
+    https://github.com/khanhas/zshelf/archive/v0.1.1.zip
+    zshelf.draft
+    zshelf.png
+)
+sha256sums=(
+    c6bc1980c90e8ea05149f3bd609667edba4e8465f9db0f6882407ca4dc70e120
+    SKIP
+    SKIP
+)
+
+image=qt:v1.1
+build() {
+    qmake zshelf.pro
+    make
+
+    # Get needed build packages
+    export DEBIAN_FRONTEND=noninteractive
+    apt-get update -y
+    apt-get install -y \
+        --no-install-recommends \
+        -o Dpkg::Options::="--force-confdef" \
+        nodejs npm
+
+    cd backend
+    npm install
+}
+
+package() {
+    install -d "$pkgdir"/opt/share/zshelf
+
+    cp -r "$srcdir"/backend "$pkgdir"/opt/share/zshelf/backend
+    cp "$srcdir"/config.json "$pkgdir"/opt/share/zshelf
+    install -D -m 755 -t "$pkgdir"/opt/share/zshelf "$srcdir"/zshelf
```

Comment:
I don't think that putting the binaries into `/opt/share/zshelf/` is a good idea. `/opt/share` is opkg's analogy to `/usr/share` which is meant to only host applications resources and not the applications itself.

Alternatives would be `/opt/zshelf/` or `/opt/lib/zshelf`. I'm not sure what is best either.

@matteodelabre @Eeems @raisjn Any ideas/notes here?

Reply:
According to Wikipedia `/usr/share`, which is what `/opt/share` is an analog to, is for
> Architecture-independent (shared) data. 

Since this isn't shared data, and is instead libraries and binaries used to run zshelf, it would make more sense to place it in `/opt/lib` which is analogous to `/usr/lib`. From Wikipedia for `/usr/lib`:
> Libraries for the binaries in /usr/bin and /usr/sbin.

Reply:
This is related to #254. I agree with what @Eeems said. Additionally, configuration files like `config.json` should go in `/opt/etc`.

Reply:
That makes sense. I will move them to `/usr/lib`.
About config.json: can I still keep it in same directory as binary and just symblink it to `/opt/etc`?

Reply:
> That makes sense. I will move them to `/usr/lib`.
Should do (`/opt/lib`*).

>About config.json: can I still keep it in same directory as binary and just symblink it to `/opt/etc`?

Sounds good. In fact I did the same [for plato](https://github.com/toltec-dev/toltec/blob/dfa23f55aab9219a4a35d6fa2efcf488b7527e73/package/plato/package#L61):

Minor nitpick: If possible, it would be cool though to have the actual config in `/opt/etc` and symlink from the binaries director to there. Maybe move the config in the packaging for that. The symlink also works as intended if the config doesn't exist in the beginning.

Reply:
It's possible but I don't see it would make any difference though.

Reply:
Did anyway 😆

Reply:
Just for the case of uninstallation to have a config stick around in /etc rather than /opt/lib. But not sure how it is handled excactly and nitpicky.